wandb/sdk/artifacts/artifact.py

```diff
@@ -1814,6 +1819,38 @@ def _download_entry(
             )
         return FilePathStr(root)
 
+    def _restore_from_cache(
+        self, root: str, path_prefix: Optional[StrPath] = None
+    ) -> bool:
+        """Restore all files available in the cache to the specified root directory.
+
+        Returns:
+            True if all files were restored, False if any entries are missing.
+        """
+        cache = get_artifact_file_cache()
+
+        def restore_entry(entry):
+            local_path = Path(root, entry.path)
+            if (
+                not self.should_download_entry(entry, path_prefix)
+                or local_path.exists()
+                and local_path.stat().st_size == entry.size
```

Comment:
is this check safe/sufficient? is this what we do in the regular downlaod code?

Reply:
This is all we do in regular download code; it would be nice to checksum them, but that would be a very significant performance cost. In a new manifest format I would like to consider storing a last modified time as a slightly better heuristic